Add ex_webrtc_endpoint

roznawsk opened the pull request.

Karolk99 reviewed (changes requested):
I think we should remove WebRTC from this PR. We did last release with previous commit and we do not wish to support it further. All changes related to the WebRTC endpoint should be merged into a separate branch dedicated to WebRTC endpoint maintenance.

roznawsk commented on engine/mix.exs:
```diff
@@ -53,15 +53,16 @@ defmodule Membrane.RTC.Engine.MixProject do
     [
-      {:membrane_core, "~> 1.1.2"},
-      {:membrane_rtp_plugin, "~> 0.28.0"},
+      {:membrane_core, "~> 1.0"},
```
Done

roznawsk commented on ex_webrtc/integration_test/README.md:
```diff
@@ -0,0 +1,16 @@
+# Integration tests
+
+Directory containing test scenarios for running integration against `membrane_rtc_engine` with the
+use of containerised headless browsers and containerised `test_videoroom` app.
```
Yes, this is the app that we use for integration tests

roznawsk commented on ex_webrtc/internal_docs/assets/ex_webrtc_endpoint_architecture.drawio.svg:
This is already updated

roznawsk commented on ex_webrtc/internal_docs/protocol.md:
Yes, the flow didn't change

roznawsk commented on ex_webrtc/lib/metrics.ex:
Currently we will have only non-RTP related metrics, such as Metadata updates

Karolk99 replied:
what about `Membrane.RTP.metrics`?

roznawsk replied:
We don't use `membrane_rtp_plugin` so we don't have any RTP metrics. We should implement them, using the `ExWebRTC.PeerConnection.get_stats()` API

Karolk99 replied:
We should sum up changes in metrics if either fishjam or cloud uses them. This way it'll be easier to adjust our cloud infrastructure to upcoming changes in membrane_rtc_engine.

roznawsk commented on examples/ex_webrtc_videoroom/README.md:
Yes it does

Karolk99 commented on ex_webrtc/README.md:
```diff
@@ -0,0 +1,32 @@
+
+[![CircleCI](https://circleci.com/gh/fishjam-cloud/membrane_rtc_engine.svg?style=svg)](https://circleci.com/gh/fishjam-cloud/membrane_rtc_engine)
+
+ExWebRTC Endpoint for the [Membrane RTC Engine](https://github.com/jellyfish-cloud/membrane_rtc_engine)
```
```suggestion
ExWebRTC Endpoint for the [Membrane RTC Engine](https://github.com/fishjam-cloud/membrane_rtc_engine)
```

Karolk99 replied:
.

Karolk99 commented on ex_webrtc/README.md:
```diff
@@ -0,0 +1,32 @@
+
+## Client SDKs
+
+- [JS/TS](https://github.com/jellyfish-dev/membrane-webrtc-js)
+- [Android](https://github.com/jellyfish-dev/membrane-webrtc-android)
+- [iOS](https://github.com/jellyfish-dev/membrane-webrtc-ios)
+- [React Native](https://github.com/jellyfish-dev/react-native-membrane-webrtc)
```
```suggestion
- [JS/TS](https://github.com/fishjam-cloud/membrane-webrtc-js)
- [Android](https://github.com/fishjam-cloud/membrane-webrtc-android)
- [iOS](https://github.com/fishjam-cloud/membrane-webrtc-ios)
- [React Native](https://github.com/fishjam-cloud/react-native-membrane-webrtc)
```

Karolk99 commented on ex_webrtc/mix.exs:
```diff
@@ -0,0 +1,153 @@
+  use Mix.Project
+
+  @version "0.1.0-dev"
+  @engine_github_url "https://github.com/jellyfish-dev/membrane_rtc_engine"
```
.

Karolk99 commented on ex_webrtc/integration_test/test_videoroom/README.md:
```diff
@@ -0,0 +1,98 @@
+# TestVideoroom
+Test application for running integration against `membrane_rtc_engine` - specifically, the WebRTC
```
.

Karolk99 reviewed (commented):
Generally there is a few requested changes and tests two tests fails

roznawsk commented on ex_webrtc/integration_test/test_videoroom/README.md:
```diff
@@ -0,0 +1,98 @@
+
+So in all tests a version for each client's sessions is maintained and gets incremented once `playwright` extracts the statistics.
+
+## Testing scenarios
```
The scenarios have been updated (removed simulcast), others are same

roznawsk commented:
Updated all occurences of `jellyfish-dev` and changed to `fishjam-cloud`. Added the project to codecov for coverage tracking. Updated badges for codecov and circleci.
